src/ca/noxid/lab/script/TscPane.java

```diff
@@ -293,6 +297,52 @@ public void caretUpdate(CaretEvent eve) {
 				}
 			}
 		});
+
+		undoManager = new UndoManager();
+		getStyledDocument().addUndoableEditListener(new UndoableEditListener() {
+			@Override
+			public void undoableEditHappened(UndoableEditEvent e) {
+			if (!e.getEdit().getPresentationName().equals("style change") && recordUndoes) {
+				System.out.println("Add edit: " + e.getEdit().getPresentationName());
```

Comment:
you should probably remove this debug statement, I know I have like more than a few lying around but this one in particular will put a lot of lines on the output.